### PR TITLE
Fix mkdocs-macros Jinja2 syntax errors in generated block docs

### PR DESCRIPTION
**Pull request by yeldarby**
## Summary

Fixes Jinja2 parse errors (`unexpected char '$'`) that occur when mkdocs-macros processes generated documentation containing `{{ $parameters.xxx }}` expressions.

## Changes

- **`build_block_docs.py`**: Added `_escape_jinja2_expressions()` helper that escapes any `{{ ... }}` blocks containing `$` signs using Jinja2 literal syntax (`{{ '{{' }}`). Applied to both block family pages and kind pages during generation.

- **`validate_docs_jinja2.py`**: New validation script that parses all `.md` files under `docs/` as Jinja2 templates and reports any syntax errors. Acts as a safety net for future regressions.

- **`docs.yml`**: Added a CI step to run the Jinja2 validation script after doc generation, before deployment.

- **`tests/docs/test_build_block_docs.py`**: Unit tests covering the escaping logic, including real-world cases (email notification descriptions, field descriptions with multiple parameter references) and edge cases (already-escaped expressions, CSS braces, bare `$` outside `{{ }}`).

### CODE REVIEW

File: tests/docs/test_build_block_docs.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
+"""Tests for the documentation build pipeline's Jinja2 escaping logic.
+
+These tests ensure that generated Markdown docs do not contain raw
+``{{ $parameters.xxx }}`` expressions that would cause mkdocs-macros
+(Jinja2) parse errors like "unexpected char '$'".
+"""
+
+import pytest
```
Comment (grzegorz-roboflow) [inline tests/docs/test_build_block_docs.py]: unused import
```suggestion
```

Comment (yeldarby) [reply tests/docs/test_build_block_docs.py]: @claude fix pls

File: development/docs/build_block_docs.py [hunk 2]
```diff
@@ -198,6 +198,11 @@ def render_template(template_name, **kwargs):
 """
 
 INLINE_UQL_PARAMETER_PATTERN = re.compile(r"({{\s*\$parameters\.(\w+)\s*}})")
+# Pattern to match any {{ ... }} containing a $ sign that hasn't already been escaped.
+# This catches patterns like {{ $parameters.xxx }} in LONG_DESCRIPTION strings,
+# Field descriptions, and other generated content that would cause mkdocs-macros
+# Jinja2 parse errors ("unexpected char '$'").
+JINJA2_DOLLAR_EXPRESSION_PATTERN = re.compile(r"(\{\{(?:(?!\}\}).)*?\$(?:(?!\}\}).)*?\}\})")
```
[the remark this answers is not in the document]
Comment (yeldarby) [reply development/docs/build_block_docs.py:205]: These should only ever be one line.

File: tests/docs/test_build_block_docs.py (hunk 1 above)
Comment (dkosowski87) [inline tests/docs/test_build_block_docs.py]: @codex please remove the unused `pytest` import in this file
